x/staking/keeper/msg_server.go

```diff
@@ -198,11 +205,6 @@ func (k msgServer) EditValidator(goCtx context.Context, msg *types.MsgEditValida
 func (k msgServer) Delegate(goCtx context.Context, msg *types.MsgDelegate) (*types.MsgDelegateResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
-	currHeight := ctx.BlockHeight()
-	if currHeight > StakingPowerUpgradeHeight {
```

Comment:
It seems we still need to fail the transaction when `currHeight > StakingPowerUpgradeHeight && currHeight < StakingPowerReverseHeight`.

If there are some `Delegate` transactions between `StakingPowerUpgradeHeight` and `StakingPowerReverseHeight`, the catch-up node will consensus failed.

Reply:
thank you.  I added a DelegatePowerRevertHeight with an approximate block height that the code might be adopted to handle the catch-up nodes.  This will be adjusted accordingly